### PR TITLE
feat: add conduit endpoints and eventsub topic

### PR DESCRIPTION
**Pull request by iProdigy**
### Prerequisites for Code Changes
* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Changes Proposed
* Add new conduits helix endpoints (CRUD conduits/shards)
* Add `conduit.shard.disabled` eventsub subscription type

### Additional Information
https://discuss.dev.twitch.com/t/available-today-twitch-chat-on-eventsub-an-api-for-sending-chat-and-the-conduit-transport-method-for-eventsub/54596#introducing-conduits-an-eventsub-transport-method-for-scale-4

Follow-up PR will create a websocket pool intended for conduit use

### CODE REVIEW

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java [hunk 1]
```diff
@@ -728,6 +729,114 @@ HystrixCommand<UpdatedDropEntitlementsList> updateDropsEntitlements(
...
+     */
+    @RequestLine("POST /eventsub/conduits")
+    @Headers({
+        "Authorization: Bearer {token}",
+        "Content-Type: application/json"
+    })
+    @Body("%7B\"shard_count\":{shard_count}%7D")
+    HystrixCommand<ConduitList> createConduit(
+        @Param("token") String authToken,
+        @Param("shard_count") int shardCount
```
Comment (PhilippHeuer) [inline rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:747]:
```suggestion
        @NotNull @Param("shard_count") int shardCount
```

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java [hunk 2]
```diff
@@ -728,6 +729,114 @@ HystrixCommand<UpdatedDropEntitlementsList> updateDropsEntitlements(
...
+    @RequestLine("PATCH /eventsub/conduits")
+    @Headers({
+        "Authorization: Bearer {token}",
+        "Content-Type: application/json"
+    })
+    @Body("%7B\"id\":\"{id}\",\"shard_count\":{shard_count}%7D")
+    HystrixCommand<ConduitList> updateConduit(
+        @Param("token") String authToken,
+        @Param("id") String conduitId,
+        @Param("shard_count") int shardCount
```
Comment (PhilippHeuer) [inline rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:782]:
```suggestion
        @NotNull @Param("id") String conduitId,
        @NotNull @Param("shard_count") int shardCount
```

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java [hunk 3]
```diff
@@ -728,6 +729,114 @@ HystrixCommand<UpdatedDropEntitlementsList> updateDropsEntitlements(
...
+     *
+     * @param authToken App access token.
+     * @param conduitId The ID of the conduit to delete.
+     * @return 204 No Content upon a successful deletion.
+     */
+    @RequestLine("DELETE /eventsub/conduits?id={id}")
+    @Headers("Authorization: Bearer {token}")
+    HystrixCommand<Void> deleteConduit(
+        @Param("token") String authToken,
+        @Param("id") String conduitId
```
Comment (PhilippHeuer) [inline rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:799]:
```suggestion
        @NotNull @Param("id") String conduitId
```

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java [hunk 4]
```diff
@@ -728,6 +729,114 @@ HystrixCommand<UpdatedDropEntitlementsList> updateDropsEntitlements(
...
+     * @param after     The cursor used to get the next page of results.
+     * @return ConduitShardList
+     */
+    @RequestLine("GET /eventsub/conduits/shards?conduit_id={conduit_id}&status={status}&after={after}")
+    @Headers("Authorization: Bearer {token}")
+    HystrixCommand<ConduitShardList> getConduitShards(
+        @Param("token") String authToken,
+        @Param("conduit_id") String conduitId,
+        @Param("status") @Nullable ShardStatus status,
+        @Param("after") @Nullable String after
```
Comment (PhilippHeuer) [inline rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:817]:
```suggestion
        @NotNull @Param("conduit_id") String conduitId,
        @Nullable @Param("status") ShardStatus status,
        @Nullable @Param("after") String after
```

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java [hunk 5]
```diff
@@ -728,6 +729,114 @@ HystrixCommand<UpdatedDropEntitlementsList> updateDropsEntitlements(
...
+     * @return ConduitShardList
+     */
+    @RequestLine("PATCH /eventsub/conduits/shards")
+    @Headers({
+        "Authorization: Bearer {token}",
+        "Content-Type: application/json"
+    })
+    HystrixCommand<ConduitShardList> updateConduitShards(
+        @Param("token") String authToken,
+        ShardsInput shards
```
Comment (PhilippHeuer) [inline rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:837]:
```suggestion
        @NotNull ShardsInput shards
```

**Review by PhilippHeuer — COMMENTED** [review]
Looks good to me, we can also add nullable/notnull to the new helix methods.

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java (hunk 1 above)
Comment (iProdigy) [reply rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:747]: agreed on all, but unnecessary for primitives

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java (hunk 2 above)
Comment (iProdigy) [reply rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java]: edited this suggestion bc of https://github.com/twitch4j/twitch4j/pull/923#discussion_r1548590223

**Review by PhilippHeuer — APPROVED** [review]

**Pull request merged**